### PR TITLE
Fix/revert to readcounter

### PR DESCRIPTION
**Pull request by lpoli**
### Changes
This PR reverts ReadMarker changes from read size to read counter

### Fixes
-

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber: https://github.com/0chain/blobber/pull/632
- 0chain: https://github.com/0chain/0chain/pull/1146
- system_test:
- zboxcli: https://github.com/0chain/zboxcli/pull/211
- zwalletcli:
- Other: ...

### CODE REVIEW

File: zboxcore/marker/readmarker.go [hunk 1]
```diff
@@ -3,9 +3,11 @@ package marker
 import (
 	"fmt"
 
+	"github.com/0chain/errors"
 	"github.com/0chain/gosdk/core/common"
 	"github.com/0chain/gosdk/core/encryption"
 	"github.com/0chain/gosdk/zboxcore/client"
+	"github.com/0chain/gosdk/zmagmacore/crypto"
```
Comment (peterlimg) [inline zboxcore/marker/readmarker.go]: Do not use packages from zmagacore, use zcncrypto instead. Create local Verify() method if necessary.

**Review by peterlimg — DISMISSED** [review]

File: zboxcore/marker/readmarker.go (hunk 1 above)
Comment (lpoli) [reply zboxcore/marker/readmarker.go]:
```
signatureHash := rm1.GetHash()
herumiScheme := zcncrypto.HerumiScheme{
PublicKey: rm1.ClientPublicKey,
}

signOK, err := herumiScheme.Verify(rm.Signature, signatureHash)
if err != nil {
return err
}
```

Is this alright?

Comment (peterlimg) [reply zboxcore/marker/readmarker.go]: I don't know how the signature was created, so I'm not sure if this is the correct way to verify it. You'd better check it. Anyway, in 0chain, we use code like:
```go
	signScheme := zcncrypto.NewSignatureScheme(scheme)
	if signScheme != nil {
		err := signScheme.SetPublicKey(publicKey)
		if err != nil {
			return false, err
		}
		return signScheme.Verify(signature, hash)
	}
```

**Review by Sriep — APPROVED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | fix/revert-to-readcounter  |
| zboxcli  | fix/revert-to-readcounter  |
| zwalletcli  | staging  |
| 0chain  | fix/revert-to-readcounter  |
| blobber  | fix/revert-to-readcounter  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by Kishan-Dhakan — APPROVED** [review]

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | fix/revert-to-readcounter  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**